### PR TITLE
Conditionally apply the patch for GP4.1.

### PR DESCRIPTION
**Pull request by haradh1**
It seems the current build system is able to apply the patch by
checking patch version, but in GPDB development environment,
it is not always true; the patch version is still 0 while the
header file has been fixed.

MADLIB-485

### CODE REVIEW

**Comment by haradh1** [comment]
Florian, please take a look and I hope you pull it shortly.

File: src/ports/greenplum/4.1/patch/Versions_4_1_0_to_4_1_1.sh.in [hunk 1]
```diff
@@ -3,7 +3,13 @@
 "@CMAKE_COMMAND@" -E copy "@GREENPLUM_4_1_SERVER_INCLUDE_DIR@/utils/syncbitvector.h" \
     server/utils/syncbitvector.h
 
-patch -p0 <<'EOF'
+# The later version has fixed this issue.
+FIXED=`perl -ne 'print "FIXED"
```
Comment (fschopp) [inline src/ports/greenplum/4.1/patch/Versions_4_1_0_to_4_1_1.sh.in]: Not a big deal but this adds perl to our build dependencies (it currently is not).

Comment (haradh1) [inline src/ports/greenplum/4.1/patch/Versions_4_1_0_to_4_1_1.sh.in]: Right, and I've never seen such environment where perl is not available except Windows.

**Comment by fschopp** [comment]
I think I'd prefer fixing (untested) by

```
sed 's/^struct SyncBitvectorContainer_s$/typedef struct SyncBitvectorContainer_s/g' \
"@GREENPLUM_4_1_SERVER_INCLUDE_DIR@/utils/syncbitvector.h" > \
server/utils/syncbitvector.h
```

**Comment by fschopp** [comment]
The above is POSIX compatible and similar to the other patches we apply.

**Comment by haradh1** [comment]
Pushed another revision, using sed.  I needed mkdir -p

**Comment by haradh1** [comment]
I'm testing with dev build, but not with released binary.

**Comment by haradh1** [comment]
OK, confirmed with GP 4.1.0.0.

File: src/ports/greenplum/4.1/patch/Versions_4_1_0_to_4_1_1.sh.in [hunk 2]
```diff
@@ -1,18 +1,6 @@
 #!/bin/sh
 
-"@CMAKE_COMMAND@" -E copy "@GREENPLUM_4_1_SERVER_INCLUDE_DIR@/utils/syncbitvector.h" \
+mkdir -p server/utils
```
Comment (fschopp) [inline src/ports/greenplum/4.1/patch/Versions_4_1_0_to_4_1_1.sh.in]: Last request (promise :)): Could you use

```
"@CMAKE_COMMAND@" -E make_directory server/utils
```

instead? That's more portable and consistent with the other patches. Thanks!

**Comment by haradh1** [comment]
Done.

**Comment by fschopp** [comment]
Merged with commit ea5562de.

**Pull request closed without merging**